### ISSUE TITLE
Searchlight issue when using (SVC + class_weights='balanced')

### ISSUE THREAD

**Issue by regilapa** [nilearn/nilearn#1639]
Hello,

I have a nilearn-based searchlight pipeline that used to run seamlessly until I tried to improve it by using:

`svc = sklearn.svm.SVC(kernel='linear', class_weight='balanced')`
`searchlight = nilearn.decoding.SearchLight(mask,  estimator=svc, radius=7.5, cv=logo)`

instead of simply:
`searchlight = nilearn.decoding.SearchLight(mask,  radius=7.5, cv=logo)`

Now, it seems to be stalled (despite no errors) and the progress displays (since yesterday): 
 _0/32108 voxels (0.00%, 263806576 seconds remaining)_

whereas the identical code (i.e., minus the svc specification) initially said:
 _0/32108 voxels (0.00%, 29000 seconds remaining)_
(and it indeed successfully finished within a few hours)

-- Any ideas of what could be causing this discrepancy?

(For the record, I have another/ROI-based pipeline that uses the svc/balanced classes without this issue, so it seems to be an interaction of using the searchlight + SVC/balanced classes command)

Thanks in advance for any thoughts,

Regina

**Comment by GaelVaroquaux** [comment]
I checked, and the searchlight uses a sklearn.svm.LinearSVC internally by
default. The solver is quite different from a SVC(kernel='linear'). This
might explain the difference.

**Comment by regilapa** [comment]
That was it, indeed--thank you!

Regina
